src/main/java/org/benetech/servicenet/adapter/DataAdapterFactory.java

```diff
@@ -17,18 +16,28 @@ public DataAdapterFactory(ApplicationContext context) {
     }
 
     public Optional<SingleDataAdapter> getSingleDataAdapter(String providerName) {
-        try {
-            return Optional.of(context.getBean(providerName + DATA_ADAPTER_SUFFIX, SingleDataAdapter.class));
-        } catch (NoSuchBeanDefinitionException | BeanNotOfRequiredTypeException e) {
+        AbstractDataAdapter adapter = getDataAdapter(providerName);
+        if (adapter instanceof SingleDataAdapter) {
+            return Optional.of((SingleDataAdapter) adapter);
```

Comment:
You should use optional.ofNullable

Reply:
I think it's not necessary, as it's already verified and can't be null (by instanceof).

Reply:
Ok